colossalai/utils/activation_checkpoint.py

```diff
@@ -70,8 +77,9 @@ def backward(ctx, *args):
 
         # Fill in inputs with appropriate saved tensors.
         for i, idx in enumerate(tensor_indices):
-            inputs[idx] = tensors[i]
-
+            tmp = tensors[i].detach().to(ctx.device)
+            tmp.requires_grad = tensors[i].requires_grad
```

Comment:
Both detach() and requires_grad setting are unnecessary.
Because inputs will be detached later in detach_variable.

Reply:
The `detach()` seems to be unnecessary, for `to()` would create a new tensor.
But the `requires_grad` is necessary, I have tested that the to() method would set `requires_grad=False` as well.

Reply:
The `detach()` seems to be unnecessary, for `to()` would create a new tensor.
But the `requires_grad` is necessary, I have tested that the to() method would set `requires_grad=False` as well.